src/main/java/org/truffleruby/core/rope/RopeNodes.java

```diff
@@ -46,9 +46,10 @@
 
 public abstract class RopeNodes {
 
+    // Preserves encoding of the top-level Rope
     @NodeChildren({
             @NodeChild(type = RubyNode.class, value = "base"),
-            @NodeChild(type = RubyNode.class, value = "offset"),
+            @NodeChild(type = RubyNode.class, value = "offset"), // in bytes
```

Comment:
I'd prefer to just call this field `byteOffset`. But I can do that in another cleanup.